File: ES_Cpp/EmbeddedSystem/API/application.cpp
#include "application.h"

using namespace std;
using namespace EmbeddedSystem::API;

Application::Application(QImage p_image, string p_name){
    m_image = p_image;
    m_name = p_name;
}

QImage Application::get_image(){
    return m_image;
}

string Application::get_name(){
    return m_name;
}

File: Embedded_Java/EmbeddedSystem/src/com/projectgame/intelligenthome/embeddedsystem/model/Application.java
package com.projectgame.intelligenthome.embeddedsystem.model;

import javafx.scene.image.Image;

/**
 * Created by Beppo-Laptop on 11/10/2015.
 */
public class Application {
    private String name;
    private Image image;
    private com.projectgame.intelligenthome.core.Application app;

    public Application(String name, Image image, com.projectgame.intelligenthome.core.Application app){
        this.name = name;
        this.image = image;
        this.app = app;
    }

    public String getName(){
        return name;
    }
    public Image getImage(){
        return image;
    }
    public com.projectgame.intelligenthome.core.Application getApp(){
        return app;
    }
}
